generate_build_config.py

```diff
@@ -33,23 +34,43 @@
 - mv $CHROOT_ROOT/build/livecd.ubuntu-cpc.* /home/ubuntu/images
 """
 
+WRITE_FILES_TEMPLATE = """\
+write_files:
+- encoding: b64
```

Comment:
I was going to suggest gz+b64 as some clouds we have seen have relatively low limits on the size of user-data.  However, I don't know of a python 2/3 compatible way to do that.  Python2 uses gzip.GzipFile on a StringIO object but Python3 would need to use io.BytesIO.

I was going to suggest (for python2):

```
     if customisation_script is not None:
         with open(customisation_script, 'rb') as f:
-            content = base64.b64encode(f.read()).decode('utf-8')
+            content_gz = StringIO.StringIO()
+            with gzip.GzipFile(fileobj=content_gz, mode='w') as fgz:
+                fgz.write(f.read())
+            content = base64.b64encode(content_gz.getvalue()).decode('utf-8')
         if content:
             output_string += '\n'
```